scc/light_client/light_client.go

```diff
@@ -0,0 +1,56 @@
+package light_client
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/0xsoniclabs/consensus/inter/idx"
+	"github.com/0xsoniclabs/sonic/scc"
+	lc_state "github.com/0xsoniclabs/sonic/scc/light_client/light_client_state"
+	"github.com/0xsoniclabs/sonic/scc/light_client/provider"
+)
+
+// LightClient is the main entry point for the light client.
+// It is responsible for managing the light client state and
+// interacting with the provider.
+type LightClient struct {
+	provider provider.Provider
+	state    *lc_state.State
+}
+
+// Config is used to configure the LightClient.
+// It requires an url for the certificate providers and an initial committee.
+type Config struct {
+	Provider string
+	Genesis  scc.Committee
+}
+
+// NewLightClient creates a new LightClient with the given config.
+// returns an error if the config does not contain a valid provider or committee.
+func NewLightClient(config Config) (*LightClient, error) {
+	if err := config.Genesis.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid committee provided: %w", err)
+	}
+	p, err := provider.NewServerFromURL(config.Provider)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create provider: %w\n", err)
+	}
+	return &LightClient{
+		state: lc_state.NewState(
+			scc.NewCommittee(
+				slices.Clone(config.Genesis.Members())...)),
```

Comment:
As discussed, some slice-cloning steps could be omitted here.

Reply:
https://github.com/0xsoniclabs/sonic/pull/120/commits/700da771b2480c34fb638f60ece472a6ed674899